riscv/decode.h

```diff
@@ -532,30 +532,27 @@ static inline bool is_overlaped(const int astart, const int asize,
 
 
 #define INT_ROUNDING(result, xrm, gb) \
-  if (gb > 0) { \
-    switch(xrm) {\
+  do { \
+    const uint64_t lsb = 1UL << (gb); \
+    const uint64_t lsb_half = lsb >> 1; \
+    switch (xrm) {\
       case VRM::RNU:\
-        result += ((uint64_t)1 << ((gb) - 1));\
+        result += lsb_half; \
         break;\
       case VRM::RNE:\
-        if ((result & ((uint64_t)0x3 << ((gb) - 1))) == 0x1){\
-            result -= ((uint64_t)1 << ((gb) - 1));\
-            }else if ((result & ((uint64_t)0x3 << ((gb) - 1))) == 0x3){\
```

Comment:
In the case that `result[gb] == 0`, the result should still be rounded upward if `(result[gb-1] != 0) && (result[gb-2:0] != 0)` (i.e., the usual round-to-nearest behavior outside of the tiebreaker).